src/state/opamEnv.ml

```diff
@@ -335,6 +335,32 @@ let is_up_to_date ?skip st =
   is_up_to_date_raw ?skip
     (updates ~set_opamroot:false ~set_opamswitch:false ~force_path:false st)
 
+(** Returns shell-appropriate statement to evaluate [cmd]. *)
+let shell_eval_invocation shell cmd =
+  match shell with
+  | SH_fish ->
+    Printf.sprintf "eval (%s)" cmd
+  | SH_csh ->
+    Printf.sprintf "eval `%s`" cmd
+  | _ ->
+    Printf.sprintf "eval $(%s)" cmd
+
+(** Returns "opam env" invocation string together with optional root and switch
+    overrides *)
+let opam_env_invocation ?root ?switch ?(set_opamswitch=false) () =
+  let root =
+    match root with
+    | None -> ""
+    | Some r -> Printf.sprintf " --root=%s" r
+  in
+  let switch =
+    match switch with
+    | None -> ""
+    | Some s -> Printf.sprintf " --switch=%s" s
+  in
```

Comment:
Slightly more concise as:
```suggestion
  let root = OpamStd.Option.map_default (Printf.sprintf " --root=%s") "" r in
  let switch = OpamStd.Option.map_default (Printf.sprintf " --switch=%s") "" s in
```

Reply:
Thanks, applied (r and s converted to "root" and "switch")